### DISCUSSION TITLE
Calculate approximately the encrypted data length in bytes based on the config used

### DISCUSSION THREAD

**Discussion by sabouaram** [ProtonMail/gopenpgp#293] [Q&A]
Hello folks,

Hope all is good, am wondering if there are some functions ... that can help calculating approximately the encrypted data length based on the config ... or in other terms is there a way in order to validate that the encrypted data has not been corrupted without the need to decrypt.

Thanks in advance :-)

Cheers !

**Comment by twiss** [comment]
Hi :wave: No, there are no functions for this.

> is there a way in order to validate that the encrypted data has not been corrupted without the need to decrypt

This isn't possible in general. You could try to parse the encrypted data as an OpenPGP message, and see if that succeeds, but there's no way to know if decryption will succeed without trying it.

**Reply by sabouaram** [reply]
Thanks a lot !

**Comment by sabouaram** [comment]
Could you explain how can I try to do the parsing without indeed decrypting.
Thanks in advance!
